src/util.js

```diff
@@ -276,6 +281,85 @@ function copyDescriptors(from, to, filter) {
     }
 }
 
+global.eventsToHandlers = global.eventsToHandlers ? global.eventsToHandlers :
+    {};
+// This function should not be used for methods that return a value
+// Using this function implies small performance overhead, should be used only
+// for debug methods
+function hookTo(prototypeObject, methodName, extension) {
+    var existingMethodImpl = prototypeObject[methodName];
+    var extensions = null;
+    if (typeof existingMethodImpl === "function") {
+        extensions = global.eventsToHandlers[methodName];
+        if (!extensions) {
+            extensions = global.eventsToHandlers[methodName]
```

Comment:
Hmm, same question from above - why do we need our own event implementation rather than fire events like in the global hooks? Did we just decide to go with local instrumentation? If so that's fine, if not we should be consistent in event handling.